src/main/java/crazypants/enderio/machine/spawner/TilePoweredSpawner.java

```diff
@@ -61,16 +67,53 @@ public void setSpawnMode(boolean isSpawnMode) {
     this.isSpawnMode = isSpawnMode;
   }
 
+  private SpawnResult reason = OK;
+  private SpawnResult oldReason = OK;
+
+  public void setReason(SpawnResult reason) {
+    this.reason = reason;
+  }
+
+  private int testAreaSeldom = 0;
+
+  public SpawnResult getReason() {
+    return reason;
+  }
+
   @Override
   protected void taskComplete() {
     super.taskComplete();
     if(isSpawnMode) {
       remainingSpawnTries = Config.poweredSpawnerSpawnCount + Config.poweredSpawnerMaxSpawnTries;
-      for (int i = 0; i < Config.poweredSpawnerSpawnCount && remainingSpawnTries > 0; ++i) {
-        if(!trySpawnEntity()) {
-          break;
+      TRIES: for (int i = 0; i < Config.poweredSpawnerSpawnCount && remainingSpawnTries > 0; ++i) {
```

Comment:
I didn't want to redo the original loop logic. And Java uses 'break' for two different things, so...